app/routes/groups_.$groupId.tsx

```diff
@@ -1,21 +1,69 @@
-import type { LoaderFunction } from '@remix-run/node';
+import type { Event } from '@prisma/client';
+import type { LoaderFunction, LoaderFunctionArgs, SerializeFrom } from '@remix-run/node';
 import { useParams, useLoaderData, Link } from '@remix-run/react';
 import { db } from '~/modules/database/db.server';
+import { EventCard } from '~/modules/events/card';
 import { json } from '@remix-run/node';
 
-export const loader: LoaderFunction = async ({ params }) => {
-  const group = await db.group.findFirstOrThrow({ where: { id: params.groupId } });
+export const loader: LoaderFunction = async ({ params }: LoaderFunctionArgs) => {
+  const group = await db.group.findFirstOrThrow({ where: { id: params.groupId }, include: { events: true } });
   return json({ group });
 };
 
+export function useDeserializeGroup() {
+  const data = useLoaderData() as SerializeFrom<typeof loader> | undefined;
+  const events = data?.group?.events;
+  if (!events) {
```

Comment:
I would just remove this! We know the loader data always has `groups with events`!